capstone/src/main/java/com/google/sps/data/AndroidManifestParser.java

```diff
@@ -1,9 +1,182 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// Link to StackOverflow reference: 
+//https://stackoverflow.com/questions/2097813/how-to-parse-the-androidmanifest-xml-file-inside-an-apk-package
 package com.google.sps.data;
 
 public class AndroidManifestParser {
+ 
+    // decompressXML -- Parse the 'compressed' binary form of Android XML docs 
+    // such as for AndroidManifest.xml in .apk files
+    public static int endDocTag = 0x00100101;
+    public static int startTag =  0x00100102;
+    public static int endTag =    0x00100103;
+    public static String spaces = "                                             ";
 
+    /*
+    Overall logic of decompressXML is to get the xml in human readable format
+    All print statement has been commented out and the only print statement is the
+    permissions in the APK
+    */
 
-    public void displayAndroidManifestContent(byte[] byteFile){
-        //I will be putting logic here, but function receives AndroidManifest.xml content
+    public void decompressXML(byte[] xml) {
+    // Compressed XML file/bytes starts with 24x bytes of data,
+    // 9 32 bit words in little endian order (LSB first):
+    //   0th word is 03 00 08 00
+    //   3rd word SEEMS TO BE:  Offset at then of StringTable
+    //   4th word is: Number of strings in string table
+    int numbStrings = LEW(xml, 4*4);
+    
+    // StringIndexTable starts at offset 24x, an array of 32 bit LE offsets
+    // of the length/string data in the StringTable.
+    int sitOff = 0x24;  // Offset of start of StringIndexTable
+    
+    // StringTable, each string is represented with a 16 bit little endian 
+    // character count, followed by that number of 16 bit (LE) (Unicode) chars.
+    int stOff = sitOff + numbStrings*4;  // StringTable follows StrIndexTable
+    
+    // XMLTags, The XML tag tree starts after some unknown content after the
+    // StringTable.  There is some unknown data after the StringTable, scan
+    // forward from this point to the flag for the start of an XML start tag.
+    int xmlTagOff = LEW(xml, 3*4);  // Start from the offset in the 3rd word.
+    // Scan forward until we find the bytes: 0x02011000(x00100102 in normal int)
+    for (int ii=xmlTagOff; ii<xml.length-4; ii+=4) {
+        if (LEW(xml, ii) == startTag) { 
+            xmlTagOff = ii;  break;
+        }
+    } 
+    
+    // XML tags and attributes:
```

Comment:
Add something at the top of this docstring that explains why it's here. Even "This is what the bytecode for AXML looks like.... "